File: main/java/org/goafabric/core/data/logic/mapper/PatientMapper.java
package org.goafabric.core.data.logic.mapper;

import org.goafabric.core.data.controller.vo.Patient;
import org.goafabric.core.data.repository.entity.PatientEo;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

import java.util.List;


@Mapper(componentModel = "jakarta", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface PatientMapper extends WorkaroundMapper {
    Patient map(PatientEo value);

    PatientEo map(Patient value);

    List<Patient> map(List<PatientEo> value);

}

File: src/main/kotlin/org/goafabric/core/organization/logic/mapper/PatientMapper.kt
package org.goafabric.core.organization.logic.mapper

import org.goafabric.core.organization.controller.dto.Patient
import org.goafabric.core.organization.persistence.entity.PatientEo
import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy

@Mapper(componentModel = "cdi", unmappedTargetPolicy = ReportingPolicy.IGNORE, uses = [AddressMapper::class])
interface PatientMapper {
    fun map(value: PatientEo): Patient
    fun map(value: Patient): PatientEo
    fun map(values: List<PatientEo>): List<Patient>
}
